### ISSUE TITLE
Request for a custom kernel

### ISSUE THREAD

**Issue by srajama1** [kokkos/kokkos-kernels#382]
Tall and skinny mat-vec A' * x where A is tall and skinny.

**Comment by srajama1** [comment]
This is for @crdohrm :)

@crdohrm : Just checking to make sure. A is sparse, right ?

**Comment by crdohrm** [comment]
No, actually A is dense. In the not too recent past, I've noticed much poorer performance of cuBLAS for transpose rather than non-transpose dense matrix products when A is tall and skinny. I think this would be very important to Belos too in their GMRES implementations where such calculations are needed.

**Comment by srajama1** [comment]
A custom kernel can be done for this one. We will take care of this.
